src/frame_info.rs

```diff
@@ -23,19 +23,32 @@ impl<S: Clone> Default for GameState<S> {
 }
 
 /// Represents an input for a single player in a single frame. The associated frame is denoted with `frame`.
-/// You do not need to create this struct, but the sessions will provide a `Vec<GameInput>` for you during `advance_frame()`.
+/// You do not need to create this struct, but the sessions will provide a `Vec<PlayerInput>` for you during `advance_frame()`.
 #[derive(Debug, Copy, Clone, PartialEq)]
 pub(crate) struct PlayerInput<I>
 where
-    I: Copy + Clone + PartialEq + bytemuck::Pod + bytemuck::Zeroable,
+    I: Copy
+        + Clone
+        + PartialEq
+        + bytemuck::NoUninit
+        + bytemuck::CheckedBitPattern
+        + bytemuck::Zeroable,
```

Comment:
Not sure PlayerInput needs all those type constraints, as tests are passing without.

Reply:
`bytemuck::Zeroable` is a key constraint used e.g. [here](https://github.com/gschup/ggrs/blob/0d70f4524b02a945999cdfed83a111e8c1140ed0/src/frame_info.rs#L46). 

The documentation on `bytemuck::NoUninit` recommends implementing `bytemuck::CheckedBitPattern` ([link](https://docs.rs/bytemuck/latest/bytemuck/trait.NoUninit.html)).